### PR TITLE
Add fuzzy search

### PR DESCRIPTION
**Pull request by nouriach**
**Add a Fuzzy Search feature to when searching Trusts by Group Name.**

1. Update the SearchGroups() method in the TrustGateway to include Contains()
2. Add new Integration Test to check for when searching by just a partial GroupName
3. Add new Integration Test to check for when searching by all parameters and a partial GroupName

### CODE REVIEW

File: TramsDataApi.Test/Integration/TrustsIntegrationTests.cs [hunk 1]
```diff
@@ -872,5 +969,17 @@ private Establishment GenerateEstablishment()
                 Qabreport = null
             };
         }
+
+        private string GenerateRandomStringSuffix()
```
Comment (RobertM905) [inline TramsDataApi.Test/Integration/TrustsIntegrationTests.cs:973]: You could probably use the NBuilder random generator for this. :)

File: TramsDataApi/Gateways/TrustGateway.cs [hunk 2]
```diff
@@ -29,10 +29,12 @@ public IList<Group> SearchGroups(string groupName, string ukprn, string companie
...
                 return _dbContext.Group.OrderBy(group => group.GroupUid).Skip((page - 1) * 10).Take(10).ToList();
             }
-            
+
+            var groupToCheck = groupName?.ToLower();
+
             return _dbContext.Group
                 .Where(g => (
-                    g.GroupName == groupName ||
+                    g.GroupName.ToLower().Contains(groupToCheck) ||
```
Comment (RobertM905) [inline TramsDataApi/Gateways/TrustGateway.cs:37]: I'm pretty sure `.Contains` is mapped to `LIKE` so is not case sensitive. 
Are you sure you need a ToLower here?

File: TramsDataApi.Test/Integration/TrustsIntegrationTests.cs [hunk 3]
```diff
@@ -379,7 +380,55 @@ public async Task ShouldReturnSubsetOfTrusts_WhenSearchingTrusts_ByGroupName()
            
             _legacyDbContext.SaveChanges();
         }
-        
+
+        [Fact]
+        public async Task ShouldReturnSubsetOfTrusts_WhenSearchingTrusts_WithFuzzySearch_ByGroupName()
+        {
+
+            var groupNamePartial = "Mygroupnam";
```
Comment (RobertM905) [inline TramsDataApi.Test/Integration/TrustsIntegrationTests.cs:388]: Might be worth having the search term have different upper and lower cases to ensure that the search is not case sensitive.

**Review by LewisDaleUK — CHANGES_REQUESTED** [review]

**Review by LewisDaleUK — APPROVED** [review]
Looks good to me :)

**Pull request merged**